### DISCUSSION TITLE
Huge form (500+ fields) with many calculated fields. How to manage it?

### DISCUSSION THREAD

**Discussion by atabachyn** [react-hook-form/react-hook-form#6598] [Q&A]
Hi,
I have very big form with many tabs / sub-tabs, dependent fields(fields from one tab are dependent on fields from another tab) and calculated fields (totals, averages).
Is it a pattern to manage form structure / consistency, all calculations, dependencies in one place?
What is the best practice to do this?
Thank you.

**Comment by bluebill1049** [comment]
i think https://react-hook-form.com/api/usewatch should be your friend to isolate the calculation.
